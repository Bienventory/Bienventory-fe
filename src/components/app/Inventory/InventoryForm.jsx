import { Button, MenuItem, TextField } from '@mui/material';
import React from 'react';
import PropTypes from 'prop-types';

const handleChange = () => {};

//pass in click handler for either get OR put route depending on parent
export default function InventoryForm(props) {
  const unitTypes = ['count', 'lbs', 'gal'];

  const { item_name, description, total_on_hand, unit_type, restock_level } =
    props.item;

  return (
    <form>
      <TextField
        id="inventory-name"
        label="Item Name"
        value={item_name}
        margin="normal"
        onChange={handleChange}
      />
      <TextField
        id="inventory-description"
        label="Description"
        value={description}
        margin="normal"
        onChange={handleChange}
      />
      <TextField
        type="number"
        id="qty-on-hand"
        label="Quantity On Hand"
        value={total_on_hand}
        margin="normal"
        onChange={handleChange}
      />

      <TextField
        id="unit-type"
        select
        label="Unit Type"
        value={unit_type}
        margin="normal"
        onChange={handleChange}
        helperText="Please select unit type"
      >
        {unitTypes.map((option) => (
          <MenuItem key={option} value={option}>
            {option}
          </MenuItem>
        ))}
      </TextField>

      <TextField
        id="restock-qty"
        label="Restock Level"
        value={restock_level}
        margin="normal"
        onChange={handleChange}
      />

      <Button style={{ display: 'block' }} variant="contained">
        Submit
      </Button>
    </form>
  );
}

InventoryForm.defaultProps = {
  item: {
    item_name: 'New Inventory Item',
    description: 'Description',
    total_on_hand: 0,
    unit_type: 'count',
    restock_level: 0,
  },
};

InventoryForm.propTypes = {
  item: PropTypes.shape({
    item_name: PropTypes.string,
    description: PropTypes.string,
    total_on_hand: PropTypes.number,
    unit_type: PropTypes.string,
    restock_level: PropTypes.number,
  }),
};
