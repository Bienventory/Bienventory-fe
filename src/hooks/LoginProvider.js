import React, { createContext, useContext, useState } from 'react';
import PropTypes from 'prop-types';

export const Context = createContext();

//context: user authentication, inventory array, menu array

const LoginProvider = ({ children }) => {
  const [user, setUser] = useState('');
  const [loggedIn, setloggedIn] = useState(false);

  const logInUser = async (googleId) => {
    // const response = await fetch(`theapi.urrl.com///${googleId}`);
    // const userObj = await response.json();
    const userObj = { google_id: googleId, notifications: false };

    setUser(userObj);
    setloggedIn(true);
  };

  const logOutUser = () => {
    setUser('');
    setloggedIn(false);
  };

  return (
    <Context.Provider value={{ user, loggedIn, logInUser, logOutUser }}>
      {children}
    </Context.Provider>
  );
};

export const uselogInUser = () => {
  const { logInUser } = useContext(Context);
  return logInUser;
};

export const uselogOutUser = () => {
  const { logOutUser } = useContext(Context);
  return logOutUser;
};

export const useloggedIn = () => {
  const { loggedIn } = useContext(Context);
  return loggedIn;
};

export const useUser = () => {
  const { user } = useContext(Context);
  return user;
};

LoginProvider.propTypes = {
  children: PropTypes.node.isRequired,
};

export default LoginProvider;
