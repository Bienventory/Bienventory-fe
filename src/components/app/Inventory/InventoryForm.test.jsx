import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter } from 'react-router';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import InventoryForm from './InventoryForm';
import inventoryDetail from '../../../fixtures/inventoryDetail.json';
import LoginProvider from '../../../hooks/LoginProvider';

const server = setupServer(
  rest.get(
    'https://bienventory.herokuapp.com/api/v1/inventory/:id',
    (req, res, ctx) => {
      return res(ctx.json(inventoryDetail));
    }
  )
);

describe('Inventory Form component', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('displays the inventory form component', async () => {
    const { container } = render(
      <LoginProvider>
        <MemoryRouter>
          <InventoryForm />
        </MemoryRouter>
      </LoginProvider>
    );

    screen.getByLabelText('Item Name');
    screen.getByLabelText('Description');
    screen.getByLabelText('Quantity On Hand');
    screen.getByLabelText('Unit Type');

    expect(container).toMatchSnapshot();
  });
});
