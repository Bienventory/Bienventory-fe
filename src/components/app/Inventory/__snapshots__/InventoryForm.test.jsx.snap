// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inventory Form component displays the inventory form component 1`] = `
<div>
  <form>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="inventory-name"
        id="inventory-name-label"
      >
        Item Name
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="inventory-name"
          type="text"
          value="New Inventory Item"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-186xcr5"
          >
            <span>
              Item Name
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="description"
        id="description-label"
      >
        Description
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="description"
          type="text"
          value="Description"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-186xcr5"
          >
            <span>
              Description
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="qty-on-hand"
        id="qty-on-hand-label"
      >
        Quantity On Hand
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="qty-on-hand"
          type="number"
          value="0"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-186xcr5"
          >
            <span>
              Quantity On Hand
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="unit-type"
        id="unit-type-label"
      >
        Unit Type
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiSelect-root css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="unit-type-label unit-type"
          class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          id="unit-type"
          role="button"
          tabindex="0"
        >
          count
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          name="unit-type"
          tabindex="-1"
          value="count"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-186xcr5"
          >
            <span>
              Unit Type
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiFormControl-root MuiFormControl-marginNormal MuiTextField-root css-1rv3ei8-MuiFormControl-root-MuiTextField-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1kty9di-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
        for="par"
        id="par-label"
      >
        Restock Level
      </label>
      <div
        class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id="par"
          type="text"
          value="0"
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-186xcr5"
          >
            <span>
              Restock Level
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
      style="display: block;"
      tabindex="0"
      type="button"
    >
      Submit
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </form>
</div>
`;
